Twilio-org/phonebank
# Release Calls Backend

## Manual Testing
1. Do `npm run db:refresh` to empty your database, recreate tables, and seed them with data.
2. Run the following curl command to authorize your user and get the user id and the a token. Note them both so you can use them in curl commands given for this process. Throughout this process assume that the number in the position of `:id` in `/users/:id` for the beginning of any given route is using this id, and [JWT TOKEN HERE] should be replaced with the token returned.
```curl
curl -X POST \
  http://localhost:3000/auth/login \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/x-www-form-urlencoded' \
  -H 'postman-token: 587da4a3-36fb-f0c9-1948-46f8d70853c3' \
  -d 'email=user%40foobar.com&password=password'
```
3. Run the following CURL command with the appropriate user id an token added to get a call and note its id:
```curl
curl -X POST \
  http://localhost:3000/users/1/campaigns/1/calls \
  -H 'authorization: JWT [JWT TOKEN HERE]' \
  -H 'cache-control: no-cache' \
  -H 'postman-token: 23d17009-325e-8b58-ab3b-8b5b9c5dc9d1'
```
4. Put the token user id, call id, and token in the appropriate place in this curl command's route and run it:
```curl
curl -X DELETE \
  http://localhost:3000/users/2/campaigns/1/calls/1 \
  -H 'authorization: JWT [JWT TOKEN HERE]' \
  -H 'cache-control: no-cache' \
  -H 'postman-token: 44c7c319-f1b2-30e7-2894-95a2822b394a'
```
Observe response status 200 with json object `{"message": "call successfully released"}` after step 4.
Additionally use `npm run db` to run postgres and `select * from calls;` check calls database mutation between each step.
